src/features/calendar/components/EventShiftModal/index.tsx

```diff
@@ -0,0 +1,697 @@
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import MapIcon from '@mui/icons-material/Map';
+import { TimePicker } from '@mui/x-date-pickers';
+import { useStore } from 'react-redux';
+import { Add, Close } from '@mui/icons-material';
+import {
+  Autocomplete,
+  Box,
+  Button,
+  Dialog,
+  IconButton,
+  TextField,
+  Typography,
+} from '@mui/material';
+import dayjs, { Dayjs } from 'dayjs';
+import { FC, useEffect, useState } from 'react';
+
+import { eventCreated } from 'features/events/store';
+import { EventsModel } from 'features/events/models/EventsModel';
+import EventTypeAutocomplete from 'features/events/components/EventTypeAutocomplete';
+import EventTypesModel from 'features/events/models/EventTypesModel';
+import LocationModal from 'features/events/components/LocationModal';
+import LocationsModel from 'features/events/models/LocationsModel';
+import messageIds from 'features/events/l10n/messageIds';
+import { useEnv } from 'core/hooks';
+import { useMessages } from 'core/i18n';
+import useModel from 'core/useModel';
+import useNumericRouteParams from 'core/hooks/useNumericRouteParams';
+import { ZetkinEventPostBody } from 'features/events/repo/EventsRepo';
+import ZUIFutures from 'zui/ZUIFutures';
+import { ZetkinEvent, ZetkinLocation } from 'utils/types/zetkin';
+
+interface EventShiftModalProps {
+  dates: [Date, Date];
+  open: boolean;
+  close: (eventShiftDate?: Date) => void;
+}
+
+const EventShiftModal: FC<EventShiftModalProps> = ({ close, dates, open }) => {
```

Comment:
I would try to break this component apart. It's not critical that we do, but it's very big right now.